components/profile.tsx

```diff
@@ -0,0 +1,277 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import { User } from "lucide-react";
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import { TabsContent, Tabs, TabsList, TabsTrigger } from "@/components/ui/tabs";
+import Image from "next/image";
+
+interface Course {
+  courseCode: string;
+  courseTitle: string;
+  section: string;
+  faculty: string;
+  credit: number;
+  totalSeat: number;
+  seatRemaining: number;
+}
+
+interface ProfileData {
+  cgpa?: number;
+  studentInfo: {
+    studentID: string;
+    fullName: string;
+    photoUrl?: string;
+    gender: string;
+    dateOfBirth: string;
+    email: string;
+    nationality: string;
+    mobileNumber: string;
+    homePhone: string;
+  };
+  programInfo: {
+    programName: string;
+    studentType: string;
+  };
+  educationalInfo: Course[];
+  guardianInfo: {
+    fatherName: string;
+    fatherOccupation: string;
+    fatherMobile: string;
+    motherName: string;
+    motherOccupation: string;
+    motherMobile: string;
+    localGuardian: string;
+    localGuardianName: string;
+    localGuardianRelation: string;
+    localGuardianPhone: string;
+    localGuardianAddress: string;
+  };
+}
+
+export default function StudentProfile() {
+  const [profileData, setProfileData] = useState<ProfileData | null>(null);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchProfileData = async () => {
+      try {
+        const res = await fetch("/api/profile", { credentials: "include" });
+        if (!res.ok) {
+          throw new Error("Network response was not ok");
+        }
+        const data = await res.json();
+        console.log("Fetched Profile Data:", data);
+        setProfileData(data);
+        setLoading(false);
+      } catch (error) {
+        console.error("Error fetching profile data:", error);
+        setLoading(false);
+      }
+    };
+
+    fetchProfileData();
+  }, []);
```

Comment:
we can just pass the data from a server component to this client component. 
For example we can call the function directly from the dashboard/profile/page.tsx instead of using a fetch call.

You can look into other page.tsx for examples.